src/DETHRACE/common/car.c

```diff
@@ -5319,7 +5794,167 @@ int CollideTwoCars(tCollision_info* car1, tCollision_info* car2, int pPass) {
     static br_vector3 oldn2;
     static int is_old_point_available;
     LOG_TRACE("(%p, %p, %d)", car1, car2, pPass);
-    NOT_IMPLEMENTED();
+
+    if (!gCar_car_collisions) {
+        return 0;
+    }
+    if (pPass < 0 && !SimpleCarCarCollisionTest(car1, car2)) {
+        return 0;
+    }
+
+    mat1 = &car1->car_master_actor->t.t.mat;
+    mat2 = &car2->car_master_actor->t.t.mat;
+    oldmat1 = &car1->oldmat;
+    oldmat2 = &car2->oldmat;
+    BrMatrix34LPInverse(&inv_oldmat1, &car1->oldmat);
+    BrMatrix34LPInverse(&inv_oldmat2, &car2->oldmat);
+    BrMatrix34Mul(&car1_to_old_car1, mat1, &inv_oldmat1);
+    BrMatrix34Mul(&car2_to_old_car2, mat2, &inv_oldmat2);
+    BrMatrix34Mul(&old_car2_to_car1, oldmat2, &inv_oldmat1);
+    GetNewBoundingBox(&new_car1_bnds, &car1->bounds[1], &car1_to_old_car1);
+    GetNewBoundingBox(&new_car2_bnds, &car2->bounds[1], &car2_to_old_car2);
+
+    for (i = 0; i < 3; ++i) {
+        ts = car1->bounds[1].min.v[i];
+        if (new_car1_bnds.min.v[i] <= ts) {
+            ts = new_car1_bnds.min.v[i];
+        }
+        new_car1_bnds.min.v[i] = ts;
+
+        ts = car1->bounds[1].max.v[i];
+        if (new_car1_bnds.max.v[i] >= ts) {
+            ts = new_car1_bnds.max.v[i];
+        }
+        new_car1_bnds.max.v[i] = ts;
+
+        ts = car2->bounds[1].min.v[i];
+        if (new_car2_bnds.min.v[i] <= ts) {
+            ts = new_car2_bnds.min.v[i];
+        }
+        new_car2_bnds.min.v[i] = ts;
+
+        ts = car2->bounds[1].max.v[i];
+        if (new_car2_bnds.max.v[i] >= ts) {
+            ts = new_car2_bnds.max.v[i];
+        }
+        new_car2_bnds.max.v[i] = ts;
```

Comment:
For this to work, you need to add a `MIN` macro definition in `macros.h`.
I'm also adding one in my pedestrian pr, so it's very useful to have.
```suggestion
        new_car1_bnds.min.v[i] = MIN(car1->bounds[1].min.v[i], new_car1_bnds.min.v[i]);
        new_car1_bnds.max.v[i] = MAX(car1->bounds[1].max.v[i], new_car1_bnds.max.v[i]);

        new_car2_bnds.min.v[i] = MIN(car2->bounds[1].min.v[i], new_car2_bnds.min.v[i]);
        new_car2_bnds.max.v[i] = MAX(car2->bounds[1].max.v[i], new_car2_bnds.max.v[i]);
```

Reply:
yes, so much cleaner 👍

Reply:
Unless you're payed using a LOC metric :D